cupy/core/core.pyx

```diff
@@ -1331,66 +1331,138 @@ divmod = create_ufunc(
 
 
 cdef _min_max_preamble = '''
+template <typename T>
 struct min_max_st{
-    type_in0_raw value;
+    T value;
     int index;
     __device__ min_max_st() : index(-1) { }
-    __device__ min_max_st(type_in0_raw v) : value(v), index(0) { }
-    __device__ min_max_st(type_in0_raw v, int i) : value(v), index(i) { }
+    __device__ min_max_st(T v) : value(v), index(0) { }
+    __device__ min_max_st(T v, int i) : value(v), index(i) { }
 };
-__device__ min_max_st my_min(const min_max_st& a, const min_max_st& b) {
+
+template <typename T>
+__device__ min_max_st<T> my_min(
+        const min_max_st<T>& a, const min_max_st<T>& b) {
+    if (a.index == -1) return b;
+    if (b.index == -1) return a;
+    return min_max_st<T>(min(a.value, b.value));
+}
+template <typename T>
+__device__ min_max_st<T> my_min_float(
+        const min_max_st<T>& a, const min_max_st<T>& b) {
+    if (a.index == -1) return b;
+    if (b.index == -1) return a;
+    if (isnan(a.value)) return a;
```

Comment:
It seems that `isnan` is introduced in Visual Studio 2015. Is it okay? We can use `_isnan` for VS.